packages/react-router/vite.config.ts

```diff
@@ -19,7 +19,7 @@ const config = defineConfig({
 export default mergeConfig(
   config,
   tanstackViteConfig({
-    entry: './src/index.tsx',
+    entry: ['./src/index.tsx', './src/ssr/client.ts', './src/ssr/server.ts'],
```

Comment:
Please use `.tsx` extensions, like what's being used for the `solid-router` build, OR switch it all to use `.ts` instead.